source/dbo/Stored Procedures/Power BI Reporting/uspMapCaseRecordChestXrayResults.sql

```diff
@@ -0,0 +1,36 @@
+﻿CREATE PROCEDURE [dbo].[uspMapCaseRecordChestXrayResults]
+AS
+BEGIN TRY
+	WITH rankedResults AS
+	(
+	SELECT mr.NotificationId
+	,mr.Result AS result
+	,ROW_NUMBER() OVER
+		(PARTITION BY mr.NotificationId
+		ORDER BY ABS(DATEDIFF(day, mr.TestDate, cd.DiagnosisDate)) ASC
+		,CASE WHEN Result = 'ConsistentWithTbCavities' THEN 1
+			WHEN Result = 'ConsistentWithTbOther' THEN 2
+			WHEN Result = 'NotConsistentWithTb' THEN 3
+			WHEN Result = 'Awaiting' THEN 4 END ASC
+		) AS rn
+	FROM [$(NTBS)].[dbo].[ManualTestResult] mr
+		INNER JOIN [dbo].[Record_CaseData] cd ON cd.NotificationId = mr.NotificationId
+	WHERE ManualTestTypeId = 4
+	)
+	UPDATE cd
+	SET ChestXRayResult =
+	CASE
+	WHEN rankedResults.result = 'Awaiting' THEN 'Awaiting'
```

Comment:
I would probably have put this in a lookup table as we have with other values we have to decode from NTBS, but I don't think this has to be done. It does mean the ranking could be stored in the same table, which would eliminate both of the CASE statements. I feel like this would be more performant at higher volumes?

Reply:
Yes, I think this would be a good idea, and probably more performant